### ISSUE TITLE
FT.CREATE TEXT WEIGHT #.# SORTABLE

### ISSUE THREAD

**Issue by blaggacao** [dragonflydb/dragonfly#3355]
From Redis Spec:
(https://redis.io/docs/latest/develop/interact/search-and-query/basic-constructs/field-and-type-options/#text-fields)
```
FT.CREATE ... SCHEMA ...
{field_name} TEXT [WEIGHT] [NOSTEM] [PHONETIC {matcher}] [SORTABLE] [NOINDEX] [WITHSUFFIXTRIE]
```

clarified by (https://redis.io/docs/latest/commands/ft.create/):

> `WEIGHT {weight}` for `TEXT` attributes, declares the importance of this attribute when calculating result accuracy. This is a multiplication factor, and defaults to 1 if not specified.

While connecting to dragonflydb:
```console
redis /run/dragonfly.sock> FT.CREATE foo ON HASH SCHEMA test TEXT WEIGHT 1.0 SORTABLE
(error) ERR Invalid field type:
```

v1.20.1

**Comment by kostasrim** [comment]
Hi @blaggacao we do not yet support full text search :(

**Comment by blaggacao** [comment]
@kostasrim any chance this specific issue could be kindly addressed to unblock my further experiements with deagonflydb?

- `WEIGHT` alone at least doesn't error
- `SORTABLE` alone at least doesn't error

I would hope that maintainers find this relatively easy to fix. It somewhat looks like an oversight such as a function signature mismatch somewhere.

**Comment by blaggacao** [comment]
Yes that would be nice!

I don't control the upstream code, in this role, I'm just an external devops.

I could carry patches, of course, but the opportunity to work with the dragonfly team seemed vastly more attractive amd thorough.

Noop is good!

**Comment by blaggacao** [comment]
Very true, as well!

Just to clarify, if I where to patch tbe python redis implementation, the thing that I'd have to "unteach" it would be `WEIGHT {weight}`? &mdash; currently it's defaulted to `1.0` in my particular client library with no flag to turn it off easily.

**Comment by romange** [comment]
yeah, also if you know how to build Dragonfly - it's like 3 lines of code to ignore `WEIGHT` - I can show you.

**Comment by blaggacao** [comment]
>  it's like 3 lines of code to ignore WEIGHT - I can show you.

Oh yes, please! I'm already patching everything [left and right](https://github.com/blaggacao/frappix/pull/6/files#diff-bf973e0a813993893693df13869e06dbdcffb0560e6e21f4d154cab1a8434b3d), so my marginal effort of carrying another patch is _very_ low :smile:

**Comment by blaggacao** [comment]
@romange Thanks a lot, that got me one step futher, indeed.

@kostasrim I think I now understand what you meant by:

> we do not yet support full text search :(

I'm not very versed in this topic, but I immediately hit the next hurdle:
```
ResponseError: unknown command `FT.ADD`
```

Likel the same for `FT.DEL` and `FT.SPELLCHECK` (which also seem to be used in the app I'm trying to run on dragonflydb).

Should we keep this issue open, or should I rather subscribe to some umbrella issue for search to get notifications?

**Comment by kostasrim** [comment]
> Should we keep this issue open, or should I rather subscribe to some umbrella issue for search to get notifications?

@dranikpg it's your call :)

**Comment by dranikpg** [comment]
> ResponseError: unknown command `FT.ADD`

Those (together with FT.DEL) are Redisearch 1.0 commands, they're deprecated a long long time ago

> FT.SPELLCHECK

We don't have full text search, so we can't provide any this functionality

---

I'm sorry, but it looks like it's all functionality that's either deprecated a long time or out of scope due to being full text search 😢 There isn't much we can do without taking up all of it

**Comment by blaggacao** [comment]
I see, thanks for the collaboration and information all of you provided!

With that, I'm happy to close this issue and defer to your internal roadmap and priorities.

Thanks again!
